server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -751,7 +751,7 @@ private boolean checkCompatibility(SchemaInfo schema, GroupProperties groupPrope
         CompatibilityChecker checker = CompatibilityCheckerFactory.getCompatibilityChecker(schema.getSerializationFormat());
 
         // Verify that the type matches the type in schemas it will be validated against.
-        if (!schemasWithVersion.stream().allMatch(x -> x.getSchemaInfo().getType().equals(schema.getType()))) {
+        if (!groupProperties.isAllowMultipleTypes() && !schemasWithVersion.stream().allMatch(x -> x.getSchemaInfo().getType().equals(schema.getType()))) {
```

Comment:
Could you elaborate on the need for this change?

Reply:
#102 was the cause for the test failures that you observed earlier. This changes fixes it. I have updated the issue description with the same as well. thanks